File: tests/fixtures/sample.ts
import { Request, Response } from 'express';

interface UserConfig {
    host: string;
    port: number;
}

class UserService {
    private db: Database;

    constructor(db: Database) {
        this.db = db;
    }

    async getUser(id: number): Promise<User | null> {
        return this.db.findById(id);
    }
}

const handler = async (req: Request, res: Response) => {
    res.json({ ok: true });
};

export function validatePort(port: number): boolean {
    return port > 0 && port <= 65535;
}

File: tests/fixtures/sample.py
import os
from typing import Optional


@dataclass
class UserService:
    db: Database

    def get_user(self, user_id: int) -> Optional[dict]:
        """Fetch a user by ID."""
        return self.db.find(user_id)

    def delete_user(self, user_id: int) -> bool:
        return self.db.remove(user_id)


def standalone_function(x: int, y: int) -> int:
    return x + y
